snakemake_executor_plugin_slurm/validation.py

```diff
@@ -6,6 +6,45 @@
 from snakemake_interface_common.exceptions import WorkflowError
 
 
+def validate_slurm_job_id(job_id, output):
+    """
+    Validate that the SLURM job ID is a positive integer.
+
+    Args:
+        job_id (str): The SLURM job ID to validate.
+        output (str): The full sbatch output to parse if job_id is invalid.
+
+    Raises:
+        WorkflowError: If the job ID is not a positive integer or we cannot
+                       determine a valid job ID from the given input string.
+    """
+    if re.match(r"^\d+$", job_id):
+        return job_id
+    else:
+        # Try matching a positive integer, raise an error if more than one match or
+        # no match found. Match standalone integers, excluding those followed by %, letters,
+        # or digits (units/percentages/floats). Allows format: "1234" or
+        # "1234; clustername" (SLURM multi-cluster format).
+
+        # If the first attempt to validate the job fails, try parsing the sbatch output
+        # a bit more sophisticatedly:
+        matches = re.findall(
+            r"\b\d+(?![%A-Za-z\d.])", output
```

Comment:
What are we capturing here? Word boundary, then some digits, then capturing something that isn't a digit or character? This may be from misunderstanding the regex but I would think we want to capture `\d+`.

Reply:
Look at your ouput: There is a xx.y%. Capturing all integers would yiel: `[<jobid>, xx, y]`. That's the thing: Admins may write whatever they want to add to the sbatch output. It may contain several numbers. We have no good reason to assume, the jobid will always be the last number. We have no good reason to assume it is the only parsable number. The regex attempts to discard all numbers followed immediately by a `%` sign, a period, or letters. I am open to suggestions.

I refrained from checking all parsable numbers against `squeue` as that would display a tremendous overhead on the SLURM control daemon. That would, however, be a secure way to validate the job IDs.

Reply:
I probably should put a verbose comment there. It's not, that I will understand the regex right away in a stressful situation or should expect the co-developers will do ...

[will be off-screen now, have a nice weekend]

Reply:
done